client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -0,0 +1,276 @@
+	return (
+		<VStack spacing={ 6 }>
+			<Notice variant="warning" density="medium">
+				<Text>
+					{ createInterpolateElement(
+						__(
+							'Before resetting your site, consider <link>exporting your content as a backup</link>.'
+						),
+						{
+							// @ts-expect-error children prop is injected by createInterpolateElement
+							link: <ExternalLink href={ `/export/${ siteDomain }` } />,
+						}
+					) }
+				</Text>
+			</Notice>
+			<Text as="p">
+				{ createInterpolateElement(
+					/* translators: <siteDomain />: site domain */
+					__(
+						"Resetting <siteDomain /> will remove all of its content but keep the site and its URL up and running. You'll also lose any modifications you've made to your current theme. This cannot be undone."
+					),
+					{
+						siteDomain: <strong>{ siteDomain }</strong>,
+					}
+				) }
+			</Text>
+			<ContentInfo siteContent={ siteContent } siteDomain={ siteDomain } />
+
+			<form onSubmit={ handleSubmit }>
+				<VStack spacing={ 4 }>
+					<DataForm< SiteResetFormData >
+						data={ formData }
+						fields={ fields }
+						form={ { type: 'regular', fields } }
+						onChange={ ( edits: { domain?: string } ) => {
+							setFormData( ( data ) => ( {
+								...data,
+								...edits,
+								domain: edits.domain?.trim() ?? data.domain,
+							} ) );
+						} }
+					/>
+					<HStack spacing={ 4 } justify="flex-end">
+						<Button variant="tertiary" onClick={ onClose }>
+							{ __( 'Cancel' ) }
+						</Button>
+						<Button
+							variant="primary"
+							type="submit"
+							isBusy={ isBusy }
+							disabled={ formData.domain !== siteDomain }
+						>
+							{ __( 'Reset site' ) }
+						</Button>
+					</HStack>
+				</VStack>
+			</form>
+		</VStack>
+	);
+}
+
+export default function SiteResetModal( { site, onClose }: { site: Site; onClose: () => void } ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const { data: siteContent } = useQuery< SiteContent >( siteResetContentSummaryQuery( site.ID ) );
+
+	const statusQuery = {
+		...siteResetStatusQuery( site.ID ),
+		...( site.is_wpcom_atomic && {
+			refetchInterval: ( query: Query< SiteResetStatus > ) => {
+				const { data } = query.state;
+
+				if ( data?.status === 'ready' || data?.status === 'completed' ) {
+					return false;
+				}
+				return 5000;
+			},
+			gcTime: 0,
+			meta: {
+				persist: false,
+			},
+		} ),
+	};
+
+	const { data: resetStatus, refetch: refetchResetStatus } =
+		useQuery< SiteResetStatus >( statusQuery );
+	const { mutate, isPending: isMutationPending } = useMutation< void, ErrorResponse >(
+		resetSiteMutation( site.ID )
+	);
+
+	const showSuccessNotice = useCallback( () => {
+		createSuccessNotice(
+			sprintf(
+				/* translators: %s: site domain */
+				__( '%s has been reset.' ),
+				site.slug
+			),
+			{ type: 'snackbar' }
+		);
+	}, [ createSuccessNotice, site.slug ] );
+
+	useEffect( () => {
```

Comment:
Would it be better to move it to the `onSuccess` of the query? This effect seems unnecessary 🤔

Reply:
Oops 🤦‍♂️